watertap/unit_models/tests/test_pressure_exchanger.py

```diff
@@ -246,6 +293,202 @@ def test_solution(self, unit_frame):
+        assert (pytest.approx(7.352e-2-7.352e-2*0.05, rel=1e-3) ==
+                value(m.fs.unit.high_pressure_outlet.flow_mass_phase_comp[0,'Liq','TDS']))
+        assert (pytest.approx(3.582e-2+7.352e-2*0.05, rel=1e-3) ==
+                value(m.fs.unit.low_pressure_outlet.flow_mass_phase_comp[0,'Liq','TDS']))
+
+    @pytest.mark.unit
+    def test_report(self, unit_frame):
+        unit_frame.fs.unit.report()
+
+class TestPressureExchanger_with_ion_prop_pack():
+    @pytest.fixture(scope="class")
+    def unit_frame(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(default={'dynamic': False})
+        m.fs.properties = property_seawater_ions.PropParameterBlock()
+        m.fs.unit = PressureExchanger(default={'property_package': m.fs.properties,\
+         'has_mass_transfer': True})
+
+        # Specify inlet conditions
+        temperature = 25 + 273.15
+
+        lowP_pressure = 101325
+        highP_mass_frac_TDS = 0.07
+        highP_pressure = 50e5
+
+        solvent_transfer=0.05
+        solute_transfer={'Na': 0.1,
+                          'Ca': 0.11,
+                          'Mg': 0.12,
+                          'SO4': 0.13,
+                          'Cl': 0.14}
+
+        feed_flow_mass = 1
+        lowP_mass_frac = {'Na': 11122e-6,
+                          'Ca': 382e-6,
+                          'Mg': 1394e-6,
+                          'SO4': 2136e-6,
+                          'Cl': 20300e-6}
+
+        highP_mass_frac = {'Na': 2*11122e-6,
+                          'Ca': 2*382e-6,
+                          'Mg': 2*1394e-6,
+                          'SO4': 2*2136e-6,
+                          'Cl':2*20300e-6}
+
+        #Specify unit
+        efficiency = 0.95
+        m.fs.unit.efficiency_pressure_exchanger.fix(efficiency)
+
+        m.fs.unit.solution_transfer_fraction[0,'H2O'].fix(solvent_transfer)
+        for s in solute_transfer:
+            m.fs.unit.solution_transfer_fraction[0, s].fix(solute_transfer[s])
+
+        m.fs.unit.low_pressure_side.properties_in[0].flow_mass_phase_comp['Liq', 'H2O'].fix(
+            feed_flow_mass * (1 - sum(x for x in lowP_mass_frac.values())))
+        for s in lowP_mass_frac:
+            m.fs.unit.low_pressure_side.properties_in[0].flow_mass_phase_comp['Liq', s].fix(feed_flow_mass * lowP_mass_frac[s])
+        m.fs.unit.low_pressure_side.properties_in[0].pressure.fix(lowP_pressure)
+        m.fs.unit.low_pressure_side.properties_in[0].temperature.fix(temperature)
+        m.fs.unit.low_pressure_side.properties_in[0].mass_frac_phase_comp
+        m.fs.unit.low_pressure_side.properties_in[0].flow_mol_phase_comp
+        m.fs.unit.low_pressure_side.properties_in[0].flow_vol
+
+        ### NOTE THE PRESSURE EXHCANGER EXPECTS EQUAL VOLUMETRIC FLOW RATES so one of mass fractions on high pressure or low pressure sides
+        # should be left unfixed, as other wise you are fixing all the mass fractions, and forcing equal flow, which leaves no degrees of freedom
```

Comment:
@avdudchenko Which mass fraction is being unfixed here as compared to before when you got the -1 DOF?

Reply:
I am leaving the high-pressure inflow mass fractions unfixed, so it can satisfy low-pressure side flow-in== high-pressure flow-out constraint (If I fix this  I get a -1DOF, as it can not balance the flow).  In the version with no mass transfer the inflow mass fractions were fixed, as the constraint ensured that both inflows were the same.